core/test/unit/migration_spec.js

```diff
@@ -190,18 +191,44 @@ describe('Migrations', function () {
             }).catch(done);
         });
 
-        it('should should only create tables, with tablesOnly setting', function (done) {
-            populate({tablesOnly: true}).then(function (result) {
-                should.exist(result);
-                result.should.be.an.Array().with.lengthOf(schemaTables.length);
+        it('should rollback if error occurs', function (done) {
+            var i = 0,
+                transaction = {
+                    commit: sandbox.stub()
+                };
 
-                createStub.called.should.be.true();
-                createStub.callCount.should.be.eql(schemaTables.length);
-                createStub.firstCall.calledWith(schemaTables[0]).should.be.true();
-                createStub.lastCall.calledWith(schemaTables[schemaTables.length - 1]).should.be.true();
-                fixturesStub.called.should.be.false();
-                done();
-            }).catch(done);
+            createStub = sandbox.stub(schema.commands, 'createTable', function () {
+                i = i + 1;
+
+                if (i > 10) {
+                    return new Promise.reject(new Error('error on table creation :('));
+                }
+
+                return new Promise.resolve();
+            });
+
+            transactionStub = sandbox.stub(db.knex, 'transaction', function (transactionStart) {
```

Comment:
Do we need to stub the transaction logic as well as the create command?

I think a transaction in knex, without any SQL, is little more than a promise wrapper. I have a hunch that this is adding complexity, and making it harder to test the code / make the code seem like it doesn't work when it is simpler?